tool/teleport/common/teleport.go

```diff
@@ -490,6 +535,255 @@ func onForward() error {
+// database agents.
+func onConfigureDatabaseBootstrap(flags configureDatabaseBootstrapFlags) error {
+	configurators, err := dbconfigurators.BuildConfigurators(flags.config)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Printf("Reading configuration at %q...\n\n", flags.config.ConfigPath)
+	if len(configurators) == 0 {
+		fmt.Println("The agent doesn’t require any extra configuration.")
+		return nil
+	}
+
+	for _, configurator := range configurators {
+		fmt.Println(configurator.Name())
+		printDBConfiguratorInstructions(configurator.Instructions())
+	}
+
+	if flags.config.Manual {
+		return nil
+	}
+
+	fmt.Print("\n")
+	confirmed, err := requestConfirmation(flags.confirm)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if !confirmed {
+		return nil
+	}
+
+	for _, configurator := range configurators {
+		results := configurator.Execute(context.TODO())
+
+		for _, result := range results {
+			printDBBootstrapResultLine(configurator.Name(), result)
+			if result.Err != nil {
+				return trace.Errorf("bootstrap failed to execute, check logs above to see the cause")
+			}
+		}
+	}
+
+	return nil
+}
+
+// printDBBootstrapResultLine prints the execution step result.
+func printDBBootstrapResultLine(bootstrapperName string, result dbconfigurators.ExecutionResult) {
+	leadSymbol := "✅"
+	endText := "done"
+	if result.Err != nil {
+		leadSymbol = "❌"
+		endText = "failed"
+	}
+
+	fmt.Printf("%s[%s] %s... %s.\n", leadSymbol, bootstrapperName, result.Description, endText)
+	if result.Err != nil {
+		fmt.Printf("Failure reason: %s\n", result.Err)
+	}
+}
+
+// printDBConfiguratorInstructions prints the database configurator
+// instructions.
+func printDBConfiguratorInstructions(instructions []dbconfigurators.Instruction) {
+	for _, instruction := range instructions {
+		fmt.Println(instruction.Description)
+		if len(instruction.Details) > 0 {
+			fmt.Println(instruction.Details)
+			fmt.Print("\n")
+		}
+	}
+}
+
+// configureDatabaseAWSFlags common flags provided to aws DB configurators.
+type configureDatabaseAWSFlags struct {
+	// types comma-separated list of database types that the policies will give
+	// access to.
+	types string
+	// typesList parsed `types` into list of types.
+	typesList []string
+	// role the AWS role that policies will be attached to.
+	role string
+	// user the AWS user that policies will be attached to.
+	user string
+	// policyName name of the generated policy.
+	policyName string
+}
+
+func (f *configureDatabaseAWSFlags) CheckAndSetDefaults() error {
+	if f.policyName == "" {
+		return trace.BadParameter("--name must be non-empty string.")
+	}
+
+	if f.types == "" {
+		return trace.BadParameter("at least one --type should be provided: %s", strings.Join(awsDatabaseTypes, ","))
+	}
+
+	f.typesList = strings.Split(f.types, ",")
+typesListLoop:
+	for _, dbType := range f.typesList {
+		for _, supportedDBType := range awsDatabaseTypes {
+			if dbType == supportedDBType {
```

Comment:
If you invert this to `!=`, you won't need the loop label.

Reply:
We need the outer loop label because the return statement is on it. We still need to `continue` the outer loop, even changing the condition. The only alternative I see here to make things easier to understand is to place in the inner loop in a separated function so that we could have:
```go
for _, dbType := range f.typesList {
	if !isSupported(dbType) {
		return trace.BadParameter("...")
	}
}
```

What do you think?

Reply:
I have a better proposal - we already have `utils.SliceContainsStr` - can we use that instead of implementing double-loop?